plugins/ndr/parser.cpp

```diff
@@ -176,6 +186,13 @@ void _ReadShaderAttribute(const UsdAttribute &attr, NdrPropertyUniquePtrVec &pro
         hints.insert({TfToken(it.first), TfStringify(it.second)});
     }
 
+    // Asset attributes have to be treated differently as they need to be considered 
+    // as strings in some parts of USD, but as assets in others. Since in practice, 
+    // these attributes always default to empty strings, it's better not to set the
+    // VtValue at all, so that we don't get errors about invalid types.
+    if (!isAsset)
+        attr.Get(&v);
```

Comment:
I am wondering if the VtValue shouldn't be initialized before passed as defaultValue, because at this stage it doesn't have a defined type

Reply:
that's the problem. These attributes are designed in a strange way in USD, being strings in some way but assets in others. If I set it as an empty asset attribute, we have an undesirable behaviour in the lookdevx UI (there's a "@@" value by default) which doesn't happen with other shader libraries like materialx. If I set it as an empty string, I get error messages about a type mismatch. The only way to get the expected behaviour is to not set the value at all, and then we get the same behaviour as all the other shader filenames